### PR TITLE
Heretics Resource: Favor

### PR DESCRIPTION
**Pull request by eeSPee**
## About The Pull Request

Heretics have a 'resource', Favor. Meaning heretics no longer gain charges in the Codex Cicatrix from sacrifices/tears in reality, but instead gain it themselves and can use it with any book. Adjusted costs and prices of reality tears. Also, heretics can read the Codex Gigas to gain more Favor, for a cost.

## Why It's Good For The Game

Further improves the heretics' abilities to function without a book. 
Allows heretics to reading the codex gigas, a stealthy way to gain favor and a more interesting objective.

## Changelog
:cl:
add: Heretics gain 'favor', replacing codex cicatrix charges.
add: Heretics can now read Codex Gigas to earn favor.
/:cl:

### CODE REVIEW

File: code/modules/library/lib_codex_gigas.dm [hunk 1]
```diff
@@ -106,3 +109,57 @@
...
+				switch(rand(1,4))
+					if (1)
+						heretic.adjustOrganLoss(ORGAN_SLOT_BRAIN,10)
+						to_chat(heretic, "<span class='warning'>Your sanity decays...</span>")
+					if (2)
+						heretic.adjustOrganLoss(ORGAN_SLOT_EYES,10)
+						to_chat(heretic, "<span class='warning'>Your eyes hurt...</span>")
+					if (3)
+						heretic.adjustOrganLoss(ORGAN_SLOT_EARS,10)
+						to_chat(heretic, "<span class='warning'>A peeping scream disturbs your concentration...</span>")
```
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "A shrill scream pains your ears..."

File: code/modules/library/lib_codex_gigas.dm [hunk 2]
```diff
@@ -106,3 +109,57 @@
...
+						to_chat(heretic, "<span class='notice'>You feel a tingle in your abdomen.</span>")
+					if (3)
+						heretic.vomit()
+						to_chat(heretic, "<span class='notice'>You feel ill...</span>")
+				
+			if (3 to 8)	//tragic
+				switch(rand(1,4))
+					if (1)
+						heretic.adjustOrganLoss(ORGAN_SLOT_BRAIN,10)
+						to_chat(heretic, "<span class='warning'>Your sanity decays...</span>")
```
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "You get a sharp headache..."

File: code/modules/library/lib_codex_gigas.dm [hunk 3]
```diff
@@ -106,3 +109,57 @@
...
+						heretic.adjustOrganLoss(ORGAN_SLOT_EARS,10)
+						to_chat(heretic, "<span class='warning'>A peeping scream disturbs your concentration...</span>")
+					if (4)
+						heretic.adjustToxLoss(10)
+						to_chat(heretic, "<span class='warning'>Your body feels weaker.</span>")
+			else			//devastating
+				switch(rand(1,6))
+					if (1)
+						heretic.adjustOrganLoss(ORGAN_SLOT_BRAIN,20)
+						to_chat(heretic, "<span class='danger'>You lose your grip with this world!</span>")
```
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "You feel a stabbing pain in your head!"

File: code/modules/library/lib_codex_gigas.dm [hunk 4]
```diff
@@ -106,3 +109,57 @@
...
+					if (2)
+						heretic.adjustOrganLoss(ORGAN_SLOT_HEART,20)
+						to_chat(heretic, "<span class='danger'>Your heart skips a beat!</span>")
```
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm]: Maybe something more descriptive like: "Your heart burns!"

File: code/modules/library/lib_codex_gigas.dm [hunk 5]
```diff
@@ -106,3 +109,57 @@
...
+					if (3)
+						heretic.adjustToxLoss(30)
+						to_chat(heretic, "<span class='danger'>Your feel a sharp pain in your abdomen!</span>")
+					if (4)
+						new /mob/living/simple_animal/hostile/netherworld/blankbody(get_turf(src))
+						to_chat(heretic, "<span class='danger'>You draw unwanted attention!</span>")
+					if (5)
+						heretic.adjustCloneLoss(15)
+						to_chat(heretic, "<span class='danger'>Your body betrays you!</span>")
```
Comment (VictorVow) [inline code/modules/library/lib_codex_gigas.dm:160]: Maybe something more descriptive like: "Your body starts to deform!"

**Review by VictorVow — DISMISSED** [review]

**Comment by VictorVow** [comment]
Cool concept and definitely hard to cheese.

**Comment by eeSPee** [comment]
Thanks for the feedback! Updated!

File: code/modules/library/lib_codex_gigas.dm (hunk 5 above)
Comment (eeSPee) [reply code/modules/library/lib_codex_gigas.dm:160]: Still thinking about this one...

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm [hunk 6]
```diff
@@ -165,7 +165,7 @@
 	gain_text = "The forgotten lords have spoken! The lord of ash have come! Fear the fire!"
 	desc = "Bring 3 corpses onto a transmutation rune, you will become immune to fire ,space ,cold and other environmental hazards and become overall sturdier to all other damages. You will gain a spell that passively creates ring of fire around you as well ,as you will gain a powerful abiltiy that let's you create a wave of flames all around you."
 	required_atoms = list(/mob/living/carbon/human)
-	cost = 3
+	cost = 9
```
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm]: You multiplied every other cost by 4 but this was 3. Was that intentional?

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm [hunk 7]
```diff
@@ -120,7 +120,7 @@
 	name = "Rustbringer's Oath"
 	desc = "Bring 3 corpses onto the transmutation rune. After you finish the ritual rust will now automatically spread from the rune. Your healing on rust is also tripled, while you become more resillient overall."
 	gain_text = "Champion of rust. Corruptor of steel. Fear the dark for Rustbringer has come!"
-	cost = 3
+	cost = 9
```
Comment (utilbird) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm]: above comment applies here too

File: code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm [hunk 8]
```diff
@@ -214,7 +214,7 @@
 	gain_text = "Man of this world. Hear me! For the time of the lord of arms has come! Emperor of Flesh guides my army!"
 	desc = "Bring 3 bodies onto a transmutation rune to gain the ability of shedding your human form, and gaining untold power."
 	required_atoms = list(/mob/living/carbon/human)
-	cost = 3
+	cost = 9
```
Comment (heepox) [inline code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm]: I do believe this is intentional that its 9 instead of 12, but i could be wrong

Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/knowledge/flesh_lore.dm]: No I think that's my mistake.

File: html/changelog.html [hunk 9]
```diff
@@ -126,6 +126,8 @@ <h3 class="author">eeSPee updated:</h3>
 				<li class="bugfix">fixed a bug with food vat not dispensing/mixing reagents properly (at all)</li>
 			</ul>
 			<h3 class="author">yyzsong updated:</h3>
+			<h2 class="date">22 November 2020</h2>
+			<h3 class="author">Archanial updated:</h3>
```
Comment (park66665) [inline html/changelog.html]: Changelog changes?

```suggestion
```

File: html/changelogs/AutoChangeLog-pr-3002.yml [hunk 10]
```diff
@@ -0,0 +1,4 @@
+author: jupyterkat
+delete-after: true
+changes:
+  - bugfix: revs and engraved messages can't get irradiated anymore
```
Comment (park66665) [inline html/changelogs/AutoChangeLog-pr-3002.yml]: You should delete this file.

**Comment by eeSPee** [comment]
I honestly don't know why github likes to stealth-update changelog.html and why there is no easy way to revert it.

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm [hunk 11]
```diff
@@ -70,7 +70,6 @@
 	name = "Curse of blindness"
 	gain_text = "Blind man walks through the world, unnoticed by the masses."
 	desc = "Curse someone with 2 minutes of complete blindness by sacrificing a pair of eyes, a screwdriver and an object that the victim has touched with their bare hands."
-	cost = 1
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm:73]: Do not remove `cost`.

File: code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm [hunk 12]
```diff
@@ -45,7 +45,6 @@
 	name = "Leeching Walk"
 	desc = "Passively heals you when you are on rusted tiles."
 	gain_text = "The strength was unparalleled, unnatural. The Blacksmith was smiling."
-	cost = 1
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/knowledge/rust_lore.dm:48]: ditto.

**Review by park66665 — COMMENTED** [review]
Also, can you remove now-obsolete `var/charge = 1` at the definition of `/obj/item/forbidden_book`?

File: code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm (hunk 11 above)
Comment (eeSPee) [reply code/modules/antagonists/eldritch_cult/knowledge/ash_lore.dm:73]: Thanks. Must have removed it by mistake when fixing conflicts.

File: code/modules/library/lib_codex_gigas.dm [hunk 13]
```diff
@@ -106,3 +109,58 @@
...
 	data["currentSection"]=currentSection
 	return data
+
+/obj/item/book/codex_gigas/proc/do_eldritch_ritual(mob/living/carbon/human/heretic)
+	var/datum/antagonist/heretic/cultie = heretic.mind.has_antag_datum(/datum/antagonist/heretic)	
+	to_chat(heretic, "<span class='notice'>You start researching the forbidden knowledge...</span>")		
+	while (cultie && do_after(heretic,10 SECONDS, src))
+		heretic.whisper(pick("hypnos","celephalis","azathoth","dagon","yig","ex oblivione","nyarlathotep","nathicana","arcadia","astrophobos"), language = /datum/language/common)
+		switch(cultie.dread)
+			if (1 || 2)	//light
```
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm]:
```suggestion
			if (1, 2)	//light
```

**Comment by bloons3** [comment]
> Further improves the heretics' abilities to function without a book.

_Why?_

**Comment by TeomanTheGreat** [comment]
> > Further improves the heretics' abilities to function without a book.
> 
> _Why?_

Validhunting sec that's why. If your book is lost your done.

File: code/modules/library/lib_codex_gigas.dm [hunk 14]
```diff
@@ -106,3 +109,58 @@
...
 	data["currentSection"]=currentSection
 	return data
+
+/obj/item/book/codex_gigas/proc/do_eldritch_ritual(mob/living/carbon/human/heretic)
+	var/datum/antagonist/heretic/cultie = heretic.mind.has_antag_datum(/datum/antagonist/heretic)	
+	to_chat(heretic, "<span class='notice'>You start researching the forbidden knowledge...</span>")		
+	while (cultie && do_after(heretic,10 SECONDS, src))
+		heretic.whisper(pick("hypnos","celephalis","azathoth","dagon","yig","ex oblivione","nyarlathotep","nathicana","arcadia","astrophobos"), language = /datum/language/common)
+		switch(cultie.dread)
+			if (1, 2)	//light
```
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm:119]:
```suggestion
			if (0 to 2)	//light
```

File: code/modules/library/lib_codex_gigas.dm [hunk 15]
```diff
@@ -27,9 +27,12 @@
 		to_chat(user, "<span class='notice'>Someone else is reading it.</span>")
 	if(ishuman(user))
 		var/mob/living/carbon/human/U = user
+		if (IS_HERETIC(U))
+			do_eldritch_ritual(U)
+			return TRUE
```
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm:32]: Codex *Gigas*? I thought it was the eldritch book: Codex Cicatrix. Is there any reason why you chose this book?

**Review by park66665 — CHANGES_REQUESTED** [review]
Screenshot 2021-02-25 221259

> **Charges** left

File: code/modules/library/lib_codex_gigas.dm [hunk 16]
```diff
@@ -106,3 +109,58 @@
 	data["name"]=currentName
 	data["currentSection"]=currentSection
 	return data
+
+/obj/item/book/codex_gigas/proc/do_eldritch_ritual(mob/living/carbon/human/heretic)
+	var/datum/antagonist/heretic/cultie = heretic.mind.has_antag_datum(/datum/antagonist/heretic)	
+	to_chat(heretic, "<span class='notice'>You start researching the forbidden knowledge...</span>")		
+	while (cultie && do_after(heretic,10 SECONDS, src))
```
Comment (park66665) [inline code/modules/library/lib_codex_gigas.dm:116]:
```suggestion
	if(do_after(heretic, 10 SECONDS, target = src))
```

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm [hunk 17]
```diff
@@ -20,7 +18,6 @@
 	. = ..()
 	if(!IS_HERETIC(user))
 		return
-	. += "The Tome holds [charge] charges."
 	. += "Use it on the floor to create a transmutation rune, used to perform rituals."
```
Comment (park66665) [inline code/modules/antagonists/eldritch_cult/eldritch_book.dm:21]:
```suggestion
```

#3450

**Comment by bloons3** [comment]
> Further improves the heretics' abilities to function without a book.

Heretics already have a mechanic to create a replacement book.
Moving the heretic points into the mind instead of the book means that there is no punishment for losing the book.
`If you lose your codex, you can make a new one by transmuting a bible, a pair of eyes, and some human skin. (better hope the chef will let you use the gibber)`
Unsure of what this PR is trying to accomplish...

**Comment by park66665** [comment]
Well, changing charges to favors only prevents unspent points being lost when a heretic loses a book. (It certainly is a small buff)

Another concern that has been raised was that the read-to-gain-favor being cheese-able albeit with a lot of efforts. I suggest a higher bracket for `dread` with even more drastic and dangerous effects, e.g. dismembering, burning the heretic, explosion, burning the book down, brain trauma, losing favor instead... you get the idea.

**Comment by eeSPee** [comment]
> Well, changing charges to favors only prevents unspent points being lost when a heretic loses a book. (It certainly is a small buff)
> 
> Another concern that has been raised was that the read-to-gain-favor being cheese-able albeit with a lot of efforts. I suggest a higher bracket for `dread` with even more drastic and dangerous effects, e.g. dismembering, burning the heretic, explosion, burning the book down, brain trauma, losing favor instead... you get the idea.

I mean, the main reason for this 'favor' to exist is so that Codex reading can exist.

Codex reading is a stealthier, less obvious alternative to reality hunting.

**Comment by park66665** [comment]
> I mean, the main reason for this 'favor' to exist is so that Codex reading can exist.

> Codex reading is a stealthier, less obvious alternative to reality hunting.

Your response is wrong at two levels:

1. You didn't reply to my second paragraph.
2. You didn't reply to my first paragraph properly either; you can make book reading work via moving book-reading feature from Codex Gigas to Codex Cicatrix, so moving charges to favor is not related to book reading. You also can search for book that the reader is holding.

You also didn't reply to my comment about Codex Gigas being used, which is counter-intuitive at best.

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm (hunk 17 above)
Comment (park66665) [reply code/modules/antagonists/eldritch_cult/eldritch_book.dm:21]: UPDATE: This will be addressed via a separate PR: #3757

**Comment by eeSPee** [comment]
> 1. You didn't reply to my second paragraph.

That's a good idea. I didn't want it to be capital crime. It's supposed to be taxing, but not seriously crippling them. Because what you suggested will leave them crippled and exposed if they make that mistake.

> 2. You didn't reply to my first paragraph properly either; you can make book reading work via moving book-reading feature from Codex Gigas to Codex Cicatrix, so moving charges to favor is not related to book reading. You also can search for book that the reader is holding.

That could work, if the heretic datum keeps tabs on the dread.
 
> You also didn't reply to my comment about Codex Gigas being used, which is counter-intuitive at best.

I thought of this as a extra way for heretics to gain resource without needing the tears in reality, or to catch up if the tears of reality are used up.

If the Codex Cicatrix does this, then every heretic will be able to 'read' risk free from the book which could work, but was not what I had in mind.

**Comment by park66665** [comment]
> That's a good idea. I didn't want it to be capital crime. It's supposed to be taxing, but not seriously crippling them. Because what you suggested will leave them crippled and exposed if they make that mistake.

Crippling the players is the point. The reason why I am asking for this is the possibility of players that would cheese the side-effects through various strategies and reading Codex Gigas til Ascension. This is going to be hilarious and not very desirable. You should ensure that Codex reading stays a non-dominant way to earn Favor: no heretics moving out to space with a Codex Gigas and returning as an ascended heretic.

I won't merge this PR at this state if you don't convince me enough of the non-necessity of higher `dread` bracket.

> If the Codex Cicatrix does this, then every heretic will be able to 'read' risk free from the book which could work, but was not what I had in mind.

It is nothing but an *easy* winner-takes-all race where librarians get a crushing home advantage that has been improvised to take read-for-favor scheme out of Codex Cicatrix. I honestly am neutral on shifting this on something other than Codex Cicatrix, but if you don't understand the implications of it, I won't look at this PR favorably. This is neither stealthy nor interesting; you should know that.

**Comment by TeomanTheGreat** [comment]
Just make it so curator can't be a heretic

**Comment by TeomanTheGreat** [comment]
> > That's a good idea. I didn't want it to be capital crime. It's supposed to be taxing, but not seriously crippling them. Because what you suggested will leave them crippled and exposed if they make that mistake.
> 
> Crippling the players is the point. The reason why I am asking for this is the possibility of players that would cheese the side-effects through various strategies and reading Codex Gigas til Ascension. This is going to be hilarious and not very desirable. You should ensure that Codex reading stays a non-dominant way to earn Favor: no heretics moving out to space with a Codex Gigas and returning as an ascended heretic.
> 
> I won't merge this PR at this state if you don't convince me enough of the non-necessity of higher `dread` bracket.
> 
> > If the Codex Cicatrix does this, then every heretic will be able to 'read' risk free from the book which could work, but was not what I had in mind.
> 
> It is nothing but an *easy* winner-takes-all race where librarians get a crushing home advantage that has been improvised to take read-for-favor scheme out of Codex Cicatrix. I honestly am neutral on shifting this on something other than Codex Cicatrix, but if you don't understand the implications of it, I won't look at this PR favorably. This is neither stealthy nor interesting; you should know that.

Just make it so curator can't be a heretic. It gives him the job of actually protecting the codex instead of just doing random shit.

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm (hunk 17 above)
Comment (Crossedfall) [reply code/modules/antagonists/eldritch_cult/eldritch_book.dm:21]:
```suggestion
	. += "The Tome holds [charge] charges."
```

**Comment by ike709** [comment]
Stale

**Pull request closed without merging**